File: src/Moryx.ControlSystem.ProcessEngine.Web/src/app/api/models/Moryx/ControlSystem/Processes/Endpoints/process-holder-group-model.ts
/* tslint:disable */
/* eslint-disable */
import { ProcessHolderPositionModel as MoryxControlSystemProcessesEndpointsProcessHolderPositionModel } from '../../../../../models/Moryx/ControlSystem/Processes/Endpoints/process-holder-position-model';
import { VisualizationModel as MoryxControlSystemProcessesEndpointsVisualizationModel } from '../../../../../models/Moryx/ControlSystem/Processes/Endpoints/visualization-model';
export interface ProcessHolderGroupModel {
  id?: number;
  isEmpty?: boolean;
  name?: string | null;
  positions?: Array<MoryxControlSystemProcessesEndpointsProcessHolderPositionModel> | null;
  visualization?: MoryxControlSystemProcessesEndpointsVisualizationModel;
}

File: src/Moryx.ControlSystem.Processes.Endpoints/Models/ProcessHolderGroupModel.cs
// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG

namespace Moryx.ControlSystem.Processes.Endpoints;

public class ProcessHolderGroupModel
{
    public long Id { get; set; }
    public string Name { get; set; }

    public bool IsEmpty => Positions.All(x => x.IsEmpty);

    public List<ProcessHolderPositionModel> Positions { get; set; } = [];

    public VisualizationModel Visualization { get; set; }
}
